staging/src/k8s.io/apiserver/pkg/server/filters/reqmgmt.go

```diff
@@ -255,17 +66,16 @@ func WithRequestManagementByClient(
 		}
 
 		for {
-			rmState := reqMgmt.curState.Load().(*RMState)
-			fs := reqMgmt.pickFlowSchema(r, rmState.flowSchemas, rmState.priorityLevelStates)
-			ps := reqMgmt.requestPriorityState(r, fs, rmState.priorityLevelStates)
-			if ps.config.Exempt {
+			rmState := reqMgmt.GetCurrentState()
+			fs := utilflowcontrol.PickFlowSchema(requestDigest, rmState.GetFlowSchemas(), rmState.GetPriorityLevelStates())
+			ps := utilflowcontrol.RequestPriorityState(requestDigest, fs, rmState.GetPriorityLevelStates())
+			if ps.IsExempt() {
 				klog.V(5).Infof("Serving %v without delay\n", r)
 				handler.ServeHTTP(w, r)
 				return
 			}
-			flowDistinguisher := reqMgmt.computeFlowDistinguisher(r, fs)
-			hashValue := reqMgmt.hashFlowID(fs.Name, flowDistinguisher)
-			quiescent, execute, afterExecute := ps.fqs.Wait(hashValue, ps.config.HandSize)
+			hashValue := utilflowcontrol.ComputeFlowDistinguisher(requestDigest, fs)
+			quiescent, execute, afterExecute := ps.GetFairQueuingSystem().Wait(hashValue, ps.GetHandSize())
```

Comment:
This also seems unnecessarily exposing details outside the package where they are homed.  Also, the hash value is only used in this call and so the hash computation could be folded in here.  Thus I suggest:
```
    quiescent, execute, afterExecute := ps.Wait(requestDigest, fs)
```

Reply:
And the early out for Exempt can be folded into Wait.  That leaves us with a sequence of three calls here (GetCurrentState, Classify, Wait) that can be merged into one.

Reply:
+1 on hiding unnecessarily details. the change is in #80956, is it?

Reply:
yes